tests/testthat/test-shinytest2-reporter.R

```diff
@@ -0,0 +1,41 @@
+testthat::test_that("e2e: reporter tab is only created when a module has reporter", {
+  app_without_reporter <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app_with_reporter <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = report_module(label = "Module with Reporter")
+  )
+
+  teal_tabs <- app_with_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    rvest::html_elements("a")
+  reporter_tabs <- setNames(
+    teal_tabs %>%
+      rvest::html_attr("data-value"),
+    teal_tabs %>%
+      rvest::html_text()
+  )
+  teal_tabs <- app_without_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    rvest::html_elements("a")
+  non_reporter_tabs <- setNames(
+    teal_tabs %>%
+      rvest::html_attr("data-value"),
+    teal_tabs %>%
+      rvest::html_text()
```

Comment:
I don't think we should use pipe on such simple 1 line calls

```suggestion
    rvest::html_attr(teal_tabs, "data-value"),
    rvest::html_text(teal_tabs)
```